q2_diversity/_procrustes.py

```diff
@@ -62,4 +68,55 @@ def procrustes_analysis(reference: OrdinationResults, other: OrdinationResults,
             sample_constraints=other.sample_constraints,
             proportion_explained=other.proportion_explained[:dimensions]
             .copy())
-    return out1, out2
+    return out1, out2, info
+
+def _procrustes_monte_carlo(reference: np.ndarray,
+                            other: np.ndarray,
+                            true_m2, 
+                            trials=1000) -> (pd.DataFrame):
+                                            
+    '''Adapted from QIIME 1's transform_coordinate_matrices
+    1. set up axes for resulting dataframe 
+    2. iterating through total # of permutations 
+    2a. shuffle rows of the other matrix 
+    2b. run procrustes analysis & store the resulting values 
+    2c. store M^2 results into final dataframe 
+
+    separate dataframe 
+    true m^2 value 
+    p value (# of m^2 trials < true m^2 / n trials)
+
+    remove dimensions from function signature & require that dimensions for ref
+    & other are the same 
+
+    '''
+    
+    df = pd.DataFrame()
+    rng = default_rng()
+
+    results = []
+    trials_below_m2 = 0
+    
+    for i in range(trials):
+
+        # shuffle rows in np array 
+        rng.shuffle(other)
+
+        # run procrustes analysis
+        _, _, m2 = procrustes(reference, other)
+        
+        # check m2 value 
+        if m2 < true_m2: 
+            trials_below_m2 += 1
+    
+    p_val = trials_below_m2 / trials
+    rounded_m2 = round(m2, 5) 
+    rounded_p_val = round(p_val, 5)
```

Comment:
For storage purposes, I think it's best to not round these numbers here. Instead q2-emperor should round the numbers to an appropriate number of decimals. What do you think?

Reply:
Sounds good! I think I would move this here: https://github.com/biocore/emperor/pull/793/commits/4e8faf46a2ce4dc5424e3c1dc7f6368db659a9a6#diff-de5ac40c027777d003fcb3f1530a2f8f35df374f22ea0e3b48377493948d3048R107